src/contracts/ChiToken.sol

```diff
@@ -119,102 +119,130 @@ contract ChiToken is IERC20, ERC20WithoutTotalSupply {
+                pop(create2(0, 0, 36, add(offset, 11)))
+                pop(create2(0, 0, 36, add(offset, 12)))
+                pop(create2(0, 0, 36, add(offset, 13)))
+                pop(create2(0, 0, 36, add(offset, 14)))
+                pop(create2(0, 0, 36, add(offset, 15)))
+                pop(create2(0, 0, 36, add(offset, 16)))
+                pop(create2(0, 0, 36, add(offset, 17)))
+                pop(create2(0, 0, 36, add(offset, 18)))
+                pop(create2(0, 0, 36, add(offset, 19)))
+                pop(create2(0, 0, 36, add(offset, 20)))
+                pop(create2(0, 0, 36, add(offset, 21)))
+                pop(create2(0, 0, 36, add(offset, 22)))
+                pop(create2(0, 0, 36, add(offset, 23)))
+                pop(create2(0, 0, 36, add(offset, 24)))
+                pop(create2(0, 0, 36, add(offset, 25)))
+                pop(create2(0, 0, 36, add(offset, 26)))
+                pop(create2(0, 0, 36, add(offset, 27)))
+                pop(create2(0, 0, 36, add(offset, 28)))
+                pop(create2(0, 0, 36, add(offset, 29)))
+                pop(create2(0, 0, 36, add(offset, 30)))
+                pop(create2(0, 0, 36, add(offset, 31)))
                 offset := add(offset, 32)
             }
-
             for {
                 let i := and(value, 0x1F)
             } i {
                 i := sub(i, 1)
             } {
-                pop(create2(0, 0, 30, offset))
+                pop(create2(0, 0, 36, offset))
                 offset := add(offset, 1)
             }
         }
-
         _mint(msg.sender, value);
         totalMinted = offset;
     }
 
-    function computeAddress2(uint256 salt) public pure returns (address child) {
-        assembly {
-            let data := mload(0x40)
-            mstore(
-                data,
-                0xff0000000000004946c0e9F43F4Dee607b0eF1fA1c0000000000000000000000
-            )
-            mstore(add(data, 21), salt)
-            mstore(
-                add(data, 53),
-                0x3c1644c68e5d6cb380c36d1bf847fdbc0c7ac28030025a2fc5e63cce23c16348
-            )
-            child := and(
-                keccak256(data, 85),
-                0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
-            )
-        }
-    }
-
     function _destroyChildren(uint256 value) internal {
+        uint256 data;
+        uint256 i;
+        uint256 end;
         assembly {
-            let i := sload(totalBurned.slot)
-            let end := add(i, value)
+            i := sload(totalBurned.slot)
+            end := add(i, value)
             sstore(totalBurned.slot, end)
 
-            let data := mload(0x40)
+            data := mload(0x40)
             mstore(
                 data,
-                0xff0000000000004946c0e9F43F4Dee607b0eF1fA1c0000000000000000000000
-            )
-            mstore(
-                add(data, 53),
-                0x3c1644c68e5d6cb380c36d1bf847fdbc0c7ac28030025a2fc5e63cce23c16348
```

Comment:
Just to see if I understood this right: this is supposed to be the hash of the deployment bytecode of the child contracts. It could be hardcoded before but you had to compute it in place since the address is not known in advance.

Reply:
This can be pre-computed in the constructor as an `immutable` so that it has a similar runtime characteristic to the mainnet `ChiToken` right?

Reply:
True, in this case the only extra cost should be the additional `MSTORE` (and, I believe, related code deployment cost) on minting.

Reply:
> so there is a very small overhead with the extra memory and hashing operations (and the extra cost on creation)

Yes, but this can be avoided by pre-computing this has in the constructor and storing it as an immutable right? There will still be a small difference as the compiler may generate a function call instead of inlining the immutable, and it has to be stored as a stack variable before being used in the `assembly` block unlike the literal. But this should be very small compared to the overall gas-cost.